### PR TITLE
Error handling improvements

### PR DESCRIPTION
**Pull request by chester-leung**
* Returns client side attestation check failures to Python. Previously, any attestation failure on the client side, e.g. hash mismatch, would print out in C++ but not be returned to Python -- Python would raise a blank XGBoost error. This PR returns a descriptive attestation error to Python, which is then raised in the form of a XGBoost error (with a description of the error).
* Returns error over RPC if thrown on server side
* Wraps up remaining C++ functions called in Python `core.py` in `API_BEGIN()` and `API_END()` calls to ensure smooth error handling. Previously, errors were either printed to console and not returned to Python or were thrown using `LOG(FATAL)` which would kill the process ungracefully.

### CODE REVIEW

File: host/src/c_api/c_api.cc [hunk 1]
```diff
@@ -933,7 +933,7 @@ bool attest_remote_report(
   // Verify the remote report to ensure its authenticity.
   result = oe_verify_report(NULL, remote_report, remote_report_size, &parsed_report);
   if (result != OE_OK) {
-    std::cout << "oe_verify_report failed " << oe_result_str(result) << std::endl;
+    LOG(FATAL) << "Remote attestation failed. Remote report verification failed.";
     goto exit;
```
Comment (podcastinator) [inline host/src/c_api/c_api.cc]: The `goto` calls are redundant because LOG(FATAL) will throw an exception.

File: host/src/c_api/c_api_mc.cc [hunk 2]
```diff
@@ -840,30 +831,30 @@ XGB_DLL int encrypt_data_with_pk(char* data, size_t len, uint8_t* pem_key, size_
...
 
   int ret;
   if((ret = mbedtls_pk_parse_keyfile( &pk, keyfile, "")) != 0) {
-    LOG(FATAL) <<"signing failed -- mbedtls_pk_parse_public_keyfile returned" << ret;
+    LOG(FATAL) << "signing failed -- mbedtls_pk_parse_public_keyfile returned " << ret;
   }
 
   ret = sign_data(pk, data, data_size, signature, sig_len);
-  return ret;
+  API_END();
```
Comment (podcastinator) [inline host/src/c_api/c_api_mc.cc:849]: Can you also add this to any other functions that can be called by the client? E.g. `encrypt_file_with_keybuf` and `decrypt_file_with_keybuf`

File: host/src/c_api/c_api_mc.cc [hunk 3]
```diff
@@ -669,7 +669,7 @@ bool attest_remote_report(
   // Verify the remote report to ensure its authenticity.
   result = oe_verify_report(NULL, remote_report, remote_report_size, &parsed_report);
   if (result != OE_OK) {
-    std::cout << "oe_verify_report failed " << oe_result_str(result) << std::endl;
+    LOG(FATAL) << "Remote attestation failed. Remote report verification failed.";
     goto exit;
```
Comment (podcastinator) [inline host/src/c_api/c_api_mc.cc]: The goto calls are redundant because LOG(FATAL) will throw an exception.

File: host/src/c_api/c_api.cc (hunk 1 above)
Comment (chester-leung) [reply host/src/c_api/c_api.cc]: Fixed

File: host/src/c_api/c_api_mc.cc (hunk 3 above)
Comment (chester-leung) [reply host/src/c_api/c_api_mc.cc]: Fixed

File: host/src/c_api/c_api_mc.cc (hunk 2 above)
Comment (chester-leung) [reply host/src/c_api/c_api_mc.cc:849]: Oops I must've missed these. Fixed

**Review by podcastinator — APPROVED** [review]

**Pull request merged**